### PR TITLE
Accept IntegerType in FHIR questionnaire for DecimalType

### LINKED ISSUE #1219: Accept integer data types in FHIR  questionnaire
**Is your feature request related to a problem? Please describe.**
When I changed the data type of the field (such as pulse oximeter) from decimal to integer in the json file(modified the screener JSON provided) , after running it I faced the below error on response submission. 

Screen Shot 2022-03-03 at 11 20 10 PM

**Describe the solution you'd like**
Accept integer data types in FHIR  questionnaire.

### PR DESCRIPTION
**Pull request by PallaviGanorkar**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1219

**Description**
Internally IntegerType is converted to DecimalType

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Bug
**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Review by aditya-07 — COMMENTED** [review]
Add tests for it.

**Comment by PallaviGanorkar** [comment]
> Add tests for it.

Yes i am working on it

File: datacapture/src/test/java/com/google/android/fhir/datacapture/mapping/ResourceMapperTest.kt [hunk 1]
```diff
@@ -869,6 +869,202 @@ class ResourceMapperTest {
     assertThat(patient.name.first().family).isEqualTo("Crawford")
   }
 
+  @Test
+  fun `extract_updateIntegerObservationForDecimalDefination_shouldUpdateAsDecimal() `() =
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/mapping/ResourceMapperTest.kt:873]: can you use a minimum test case?

**Review by jingtang10 — CHANGES_REQUESTED** [review]

**Review by jingtang10 — APPROVED** [review]
Thanks Pallavi! 🙏

**Pull request merged**